core/pkg/server/sender.go

```diff
@@ -1265,21 +1265,31 @@ func (s *Sender) sendRequestLogArtifact(record *service.Record, msg *service.Log
 func (s *Sender) sendRequestDownloadArtifact(record *service.Record, msg *service.DownloadArtifactRequest) {
 	var response service.DownloadArtifactResponse
 
-	if err := artifacts.NewArtifactDownloader(
-		s.ctx,
-		s.graphqlClient,
-		s.fileTransferManager,
-		msg.ArtifactId,
-		msg.DownloadRoot,
-		msg.AllowMissingReferences,
-		msg.SkipCache,
-		msg.PathPrefix,
-	).Download(); err != nil {
-		s.logger.CaptureError(
-			fmt.Errorf("sender: failed to download artifact: %v", err))
-		response.ErrorMessage = err.Error()
+	// Perform the nil check on the GraphQL client if in offline mode.
+	if s.graphqlClient == nil {
+		// Offline mode handling:
+		s.logger.Error("sender: sendRequestDownloadArtifact: cannot download artifact in offline mode")
+		response.ErrorMessage = "Artifact downloads are not supported in offline mode. " +
+			"Run 'wandb sync' after the run completes to sync the request."
```

Comment:
you could just respond here and return early, avoiding the nesting

Reply:
I followed the style used in `sendRequestStopStatus` which does a similar thing.

Reply:
Ah yeah we are not consistent. If you like this one better it’s ok, just a matter of preference.

tho, if you use this one, probably should fix the if else

Reply:
Sounds good, will fix that for readability.